javascript/javascript3/week3/ShoppingCartUsingClasses/main.js

```diff
@@ -0,0 +1,115 @@
+//Shopping Cart Homework js3/week3
+class Product {
+  constructor(name, price) {
+    this.name = name;
+    this.price = price;
+  }
+
+  convertToCurrency(currency) {
+    if (currency.toLowerCase().includes("dollar")) {
+      return (this.price / 6.27).toFixed(2);
+    } else if (currency.toLowerCase().includes("euro")) {
+      return (this.price / 7.44).toFixed(2);
+    } else if (currency.toLowerCase().includes("pound")) {
+      return (this.price / 8.67).toFixed(2);
+    } else {
+      return this.price;
+    }
+  }
+}
+
+class ShoppingCart {
+  constructor() {
+    this.products = [];
+  }
+
+  addProduct(product) {
+    this.products.push(product);
+  }
+
+  removeProduct(product) {
+    const deleteItem = this.products.filter(
+      (item) => product.name !== item.name
+    );
+    this.products = deleteItem;
+  }
+  searchProduct() {
+    const input = document.querySelector("#search-box").value;
+    if (input) {
+      const found = this.products.filter((item) => item.name === input);
+      console.log(`Found item is ${found}`);
+    } else {
+      console.log("search product not found");
+    }
+  }
+
+  getTotal() {
+    const totalPrice = this.products.reduce((x, y) => x + y.price, 0);
+    return totalPrice;
+  }
+
+  renderProducts() {
+    this.products.forEach((item) => {
+      const productDetails = document.createElement("li");
+      productDetails.innerHTML = "Product Details: ";
+      document.body.appendChild(productDetails);
+
+      const productName = document.createElement("p");
+      productName.innerHTML = `Product name: ${item.name}`;
+      document.body.appendChild(productName);
+
+      const price = document.createElement("p");
+      price.innerHTML = `Price of the product: ${item.price}`;
+      document.body.appendChild(price);
+    });
+  }
+
+  getUser() {
+    fetch("https://jsonplaceholder.typicode.com/users/1")
+      .then((res) => res.json())
+      .then((data) => {
+        console.log(data);
+        const userDiv = document.createElement("div");
+        document.body.appendChild(userDiv);
+        const userName = document.createElement("h4");
+        userDiv.appendChild(userName);
+        userName.innerHTML = `UserName: ${data.username}`;
+        const name = document.createElement("h4");
+        userDiv.appendChild(name);
+        name.innerHTML = `Full name: ${data.name}`;
+      })
+      .then(() => {
+        const totalPrice = document.createElement("h2");
+        totalPrice.innerHTML = `Total price =  ${this.getTotal()}`;
+        document.body.appendChild(totalPrice);
+      });
+  }
+}
+
+const shoppingCart = new ShoppingCart();
+const search = document.querySelector("#search-btn");
+search.addEventListener("click", shoppingCart.searchProduct());
```

Comment:
The `shoppingCart.searchProduct()` function is runing when you run the `html` file and that's because you are declaring it and calling it Immediately. 

To fix this, you can for example run as a callback function as the following:

```suggestion
  search.addEventListener("click", ()=>shoppingCart.searchProduct());
```

Reply:
ok thanks a lot

Reply:
i have updated this ! thanks